src/tools/gcc.lua

```diff
@@ -93,9 +105,11 @@
 			High = "-Wall",
 			Off = "-w",
 		},
-		symbols = {
-			On = "-g"
-		},
+		symbols = function(cfg, mappings)
+			local values = gcc.getdebugformat(cfg)
+			local debugformat = values[cfg.debugformat] or ""
+			return { On = "-g" .. debugformat }
```

Comment:
Is it a supported feature that we return a table from this function?
I thought this function was just meant to return the options string (the value part)...?

Assuming this works, it should also support `FastLink` and `Full` values for `symbols`. Docs allege that if they are undefined for a platform, they are a synonym for `On`.

Reply:
Returning a table from this function is to preserve as much of the existing behavior prior to this change as possible.

The test for 'function' type and executing the function [here](https://github.com/premake/premake-core/pull/1067/files/95d1c021f6345558291aba7ea71530f9b94eba37#diff-00c8206c40c5891453b9c823076a1c54R495) is what triggers this logic to take place.

Reply:
Ah. I see what you mean about `FastLink` and `Full`. Will add those in when I get home later.

Reply:
How do I add a default value to the map so I can future-proof these values? Specifically I am referring to [this](https://github.com/premake/premake-core/pull/1067/files/95d1c021f6345558291aba7ea71530f9b94eba37#diff-00c8206c40c5891453b9c823076a1c54R524) check. I do not know what I would google for in order to get map._ working.